### ISSUE TITLE
React-Aria Switch component: Screen readers do not read out the changed value when using a mouse, but do with keyboard.

### ISSUE THREAD

**Issue by EricBPower** [adobe/react-spectrum#5791]
### Provide a general summary of the issue here

I've used the React-Aria tailwind starter as a base for my use of React-Aria Switch component.  
In that as well as just in the Component documentation pages you provide online, I noticed that keyboard support for the switch component correctly reads out changed values when using keyboard navigation to focus and toggle the switch component.  However clicking on the Switch component with a mouse is not reading off the correct changed checked state.

### 🤔 Expected Behavior?

The identical screen reader support for reading out toggled checked/unchecked state should be applied to mouse clicks as exists for keyboard navigation.

### 😯 Current Behavior

Clicking on the switch component with a mouse is reading out the checked state prior that was set prior to clicking on it and never reading out the changed checked state.

But doing the same using keyboard navigation is doing it correctly.

### 🖥️ Steps to Reproduce
Screenreader: https://chromewebstore.google.com/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn?pli=1

1. visit your example page:  https://react-spectrum.adobe.com/react-aria/Switch.html#example
2. With a screen reader turned on, using a mouse click on the switch in the example. It will read out the original switch checked state instead of what you have changed it to.  Click the switch control again and no screen reader is read out at all.

Now do the same using keyboard navigation:
1. With screen reader turned on, tab to the switch component (current value is read out correctly)
2. Use the keyboard SPACE key to toggle the switch (the screen reader reads out the NEW checked state "checked" or "not checked".  Hit SPACE again and the new correct checked state is read out again.
3. This is the expected CORRECT behavior that we also expect on mouse clicks.

### Version

"react-aria": "^3.31.1",     "react-aria-components": "^1.0.1",

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

Windows 11 also tested on MacOS

### 🧢 Your Company/Team

DHI

**Comment by LFDanLu** [comment]
This seems to be specific to this particular screenreader as the switch's state is properly announced in VoiceOver, NVDA, and JAWS from a quick testing sweep. This issue occurs even with the [WAI ARIA APG](https://www.w3.org/WAI/ARIA/apg/patterns/switch/examples/switch/) example too, so it seems to be non-specific to our implementation. I'm not sure how this performs with ChromeVox though, I don't own a ChromeOS device myself unfortunately.

**Comment by EricBPower** [comment]
Thank you
